src/Symfony/Component/Notifier/Bridge/Lox24/Webhook/LOX24RequestParser.php

```diff
@@ -33,28 +45,148 @@ protected function getRequestMatcher(): RequestMatcherInterface
         $name = $payload['name'] ?? null;
         $data = $payload['data'] ?? [];
 
-        if ('sms.delivery' !== $name) {
-            throw new RejectWebhookException(400, 'Notification name is not \'sms.delivery\'');
+        // Validate that we support this event type
+        if (!\in_array($name, self::SUPPORTED_EVENTS, true)) {
+            throw new RejectWebhookException(400, \sprintf('Unsupported LOX24 event type: "%s".', $name));
         }
 
-        if (!isset($data['id'], $data['status_code'])) {
-            throw new RejectWebhookException(406, 'Payload is malformed.');
+        // Validate basic payload structure
+        if (!isset($payload['id'], $payload['name'], $data)) {
+            throw new RejectWebhookException(406, 'Payload is malformed - missing required fields.');
         }
 
-        $code = $data['status_code'];
+        return $this->createEventFromPayload($name, $data, $payload);
+    }
+
+    private function createEventFromPayload(string $eventName, array $data, array $payload): SmsEvent|RemoteEvent|null
+    {
+        switch ($eventName) {
+            case 'sms.delivery':
+            case 'sms.delivery.dryrun':
+                return $this->createDeliveryEvent($data, $payload);
+
+            case 'sms.incoming':
+                return $this->createIncomingSmsEvent($data, $payload);
+
+            case 'sms.email.parsing.success':
+            case 'sms.email.parsing.fail':
+                return $this->createEmailParsingEvent($eventName, $data, $payload);
+
+            case 'phone.subscribe':
+            case 'phone.unsubscribe':
+                return $this->createSubscriptionEvent($eventName, $data, $payload);
 
-        if (0 === $code) {
+            case 'fraud-check.done':
+                return $this->createFraudCheckEvent($data, $payload);
+
+            default:
+                // This shouldn't happen due to the validation above, but handle gracefully
+                return new RemoteEvent($eventName, $payload['id'], $payload);
+        }
+    }
+
+    private function createDeliveryEvent(array $data, array $payload): ?SmsEvent
+    {
+        if (!isset($data['id'])) {
+            throw new RejectWebhookException(406, 'Delivery event payload is malformed - missing "id".');
+        }
+
+        // LOX24 can send either status_code or dlr_code (or both)
+        $statusCode = $data['status_code'] ?? null;
+        $dlrCode = $data['dlr_code'] ?? null;
+
+        // If we have dlr_code, use it (more specific delivery status)
+        if (null !== $dlrCode) {
+            return $this->createEventFromDlrCode($dlrCode, $data, $payload);
+        }
+
+        // Fallback to status_code if no dlr_code
+        if (null !== $statusCode) {
+            return $this->createEventFromStatusCode($statusCode, $data, $payload);
+        }
+
+        throw new RejectWebhookException(406, 'Delivery event payload is malformed - missing "status_code" or "dlr_code".');
```

Comment:
```suggestion
        throw new RejectWebhookException(406, 'Missing "status_code" or "dlr_code".');
```

Let's tweak the message to only mention the field that is missing.